### DISCUSSION TITLE
Bluetooth Proxy?

### DISCUSSION THREAD

**Discussion by thesebastianf** [Blackymas/NSPanel_HA_Blueprint#206] [Ideas]
Hey 
is it possible to also / additonally use the NS Panel as bluetooth proxy device https://esphome.io/components/bluetooth_proxy.html

Thanks if anybody know this

**Comment by deejaybeam** [comment]
Yes this would be possible, but the esp has limited ressources. So we decided not to enable this feature, because if esphome becomes too big, ota is not possible anymore.

**Comment by marcosscriven** [comment]
It’s quite easy config to add to the ESP yaml, without conflicting with the yaml from this project. 

That said, at least in UK, back boxes are metal, and so not an ideal place to be receiving BLE anyway.

**Reply by deejaybeam** [reply]
This is correct, just a few lines in yaml, but when compiling it adds more libs and increases the flash size of the firmware.

**Comment by fuzzysb** [comment]
I have had major issues with the NSPanel's when Bluetooth proxy is enabled even with a minimal custom config. the device will reboot at regular intervals when enabled which is undesirable. Every time there has been a BT proxy update, i have been retesting. but have the same issue. this is also the case on the Sonoff M5 switches i have also

**Reply by ghost** [reply]
I think that's been a common experience. I'd be hesitant to enable it on nspanel. You've also got a TFT between the esp32 and the room in most cases so may not give the best results even if it were stable

**Comment by skrashevich** [comment]
it's real with esp-idf framework, _sdkconfig_options: CONFIG_COMPILER_OPTIMIZATION_SIZE: y_
and disabled http server....

**Reply by edwardtfn** [reply]
v4.1 is coming with support to esp-idf and no dependancies to `web_server`, but I don't wanna say it will be compatible with `bluetooth_proxy` as this requires lots of tests.
You probably will be able to enable it, but as that component takes a huge chunk of memory, I don't know how stable your system will be.
